eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/ReplicationEndpoint.scala

```diff
@@ -216,8 +216,9 @@ object ReplicationEndpoint {
  * @param logFactory Factory of log actor `Props`. The `String` parameter of the factory is a unique
  *                   log id generated by this endpoint. The log actor must be assigned this log id.
  * @param connections Replication connections to other replication endpoints.
- * @param filters Replication filters applied locally. Filters are applied to individual
- *                event logs where filter keys are the corresponding event log names.
+ * @param filters Replication filters applied locally. Filters are applied when replicating to
+ *               individual target logs (keys are log ids of remote logs) or from individual
+ *               local logs to all corresponding targets logs (keys are log-names).
```

Comment:
I think we should have two filter maps here: one for maintaining filters per (local/source) log name (`sourceFilters`) and another one for maintaining filter per (remote/target) log id (`targetFilters`). This approach doesn't mix map keys with different semantics in a single map.

In addition to replacing a source filter with a target filter, if defined, we should also support use cases where a source filter is AND-combined with a target filter. An example is to have a source filter that separates "private" events (that should never get replicated) from public events (that can be replicated to other locations).